tracer/src/Datadog.Trace/Debugger/ExceptionAutoInstrumentation/ExceptionTrackManager.cs

```diff
@@ -288,6 +299,14 @@ private static void ProcessException(Exception exception, ErrorOriginKind errorO
 
                         if (!allFrames[frameIndex--].Method.Equals(frame.MethodInfo.Method))
                         {
+                            var participatingFrame = allFrames[frameIndex + 1];
```

Comment:
NIT: decrementing the index then reincremeting it afterwards is hard to reason about.

Maybe refactor to something like:

```csharp
ref var participatingFrame = ref allFrames[frameIndex];
frameIndex -= 1;

if (!participatingFrame.Method.Equals(frame.MethodInfo.Method))
{
    var noCaptureReason = GetNoCaptureReason(participatingFrame, @case.Probes.FirstOrDefault(p => p.Method.Equals(participatingFrame.MethodIdentifier)));
```

Reply:
Done in 753b806c79ee4ed635377cbe17dec51ba5641c29.